### PR TITLE
fix: export apisix type for encoding/json package

### LINKED ISSUE #912: bug: Structured log field types are not uniform
### Issue description

ApisixRoute update event arrived, new object and old object are empty in log.

```
{"level":"debug","time":"2022-03-09T14:19:23.622142136+08:00","caller":"ingress/apisix_route.go:378","message":"ApisixRoute update event arrived","new object":{},"old object":{}}
```

Inconsistent data types will cause elasticsearch write failures

### Environment

- your apisix-ingress-controller version (output of apisix-ingress-controller version --long):
Version: master branch
Git SHA: 1159522
Go Version: go1.17.2
Building OS/Arch: linux/amd64
Running OS/Arch: linux/amd64
- your Kubernetes cluster version (output of kubectl version):
- if you run apisix-ingress-controller in Bare-metal environment, also show your OS version (uname -a):

### Minimal test code / Steps to reproduce

maybe the apisixRoute fields are not exported?
```
type apisixRoute struct {
	groupVersion string
	v2beta1      *configv2beta1.ApisixRoute
	v2beta2      *configv2beta2.ApisixRoute
	v2beta3      *configv2beta3.ApisixRoute
}
```

### Actual result

#

### Error log

#

### Expected result

_No response_

### PR DESCRIPTION
**Pull request by Belyenochi**
### Type of change:

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:
Resolved #912 
The encoding/json package and similar packages ignore unexported fields, so we should export these type.

### CODE REVIEW

**Comment by Belyenochi** [comment]
hi @tao12345666333 , I tried doing some export renames and fixes for some type issues, I'm not sure if my naming is in line with the project specification, could you please give me some advice.

File: pkg/kube/apisix_plugin_config.go [hunk 1]
```diff
@@ -22,14 +22,14 @@ import (
 )
 
 const (
-	// ApisixPluginConfigV2beta3 represents the ApisixPluginConfig in apisix.apache.org/v2beta3 group version
-	ApisixPluginConfigV2beta3 = "apisix.apache.org/v2beta3"
+	// ApisixPluginConfigV2beta3 represents the ApisixPluginConfig in apisix.apache.org/GetV2beta3 group version
+	ApisixPluginConfigV2beta3 = "apisix.apache.org/GetV2beta3"
```
Comment (tao12345666333) [inline pkg/kube/apisix_plugin_config.go:26]: I don't think it should be modified here

File: pkg/kube/ingress.go [hunk 2]
```diff
@@ -26,22 +26,22 @@ import (
...
 
 const (
-	// IngressV1 represents the Ingress in networking/v1 group version.
-	IngressV1 = "networking/v1"
-	// IngressV1beta1 represents the Ingress in networking/v1beta1 group version.
-	IngressV1beta1 = "networking/v1beta1"
-	// IngressExtensionsV1beta1 represents the Ingress in extensions/v1beta1 group version.
-	IngressExtensionsV1beta1 = "extensions/v1beta1"
+	// IngressV1 represents the Ingress in networking/V1 group version.
+	IngressV1 = "networking/V1"
```
Comment (tao12345666333) [inline pkg/kube/ingress.go:30]: these are also

**Review by tao12345666333 — CHANGES_REQUESTED** [review]

**Comment by tao12345666333** [comment]
All "apisix.apache.org/V2beta3" needs to remain as they are, since this is the version information we use

**Comment by Belyenochi** [comment]
> All "apisix.apache.org/v2beta3" needs to remain as they are, since this is the version information we use

I totally agree, it was my omission.

**Review by tao12345666333 — DISMISSED** [review]
LGTM

File: pkg/kube/apisix_route.go [hunk 3]
```diff
@@ -54,21 +54,21 @@ type ApisixRouteInformer interface {
 // ApisixRoute is an encapsulation for ApisixRoute resource with different
 // versions, for now, they are apisix.apache.org/v1 and apisix.apache.org/v2alpha1
 type ApisixRoute interface {
-	// GroupVersion returns the api group version of the
+	// GetGroupVersion returns the api group version of the
```
Comment (tokers) [inline pkg/kube/apisix_route.go:57]: Frankly, these changes are trivial and useless, I suggest reverting them back, or it'll also pollute the result of git blame.

Comment (Belyenochi) [reply pkg/kube/apisix_route.go:57]: @tokers  Thanks for your suggestion, I have another optimization idea, we could provide factory function that returns a concrete implementation runtime.Object for the given GVK instand of kube/apisix_route.go(or ingress.go etcd).

Comment (tokers) [reply pkg/kube/apisix_route.go:57]: That sounds good!

**Review by gxthrj — APPROVED** [review]

File: pkg/kube/apisix_route.go (hunk 3 above)
Comment (tao12345666333) [reply pkg/kube/apisix_route.go:57]: > @tokers Thanks for your suggestion, I have another optimization idea, we could provide factory function that returns a concrete implementation runtime.Object for the given GVK instand of kube/apisix_route.go(or ingress.go etc).

@Belyenochi Will you update this PR based on this idea?

**Comment by Belyenochi** [comment]
@tao12345666333 tao, I'm trying to simplify this module, will update the commit in two weeks.

File: pkg/kube/apisix_route.go (hunk 3 above)
Comment (Belyenochi) [reply pkg/kube/apisix_route.go:57]: > > @tokers Thanks for your suggestion, I have another optimization idea, we could provide factory function that returns a concrete implementation runtime.Object for the given GVK instand of kube/apisix_route.go(or ingress.go etc).
> 
> @Belyenochi Will you update this PR based on this idea?

tao, I'm trying to simplify this module, will update the commit in two weeks.

Comment (tao12345666333) [reply pkg/kube/apisix_route.go:57]: Thanks!

Comment (tao12345666333) [reply pkg/kube/apisix_route.go:57]: @Belyenochi do you have time to pick this up?

**Pull request closed without merging**